pwiz_tools/Skyline/SettingsUI/EditCustomMoleculeDlg.cs

```diff
@@ -678,6 +681,24 @@ public void OkDialog()
                 if (!_formulaBox.ValidateMonoText(helper))
                     return;
             }
+
+            // If editing the molecule, make sure that the formula makes sense with any child precursors (e.g. not removing more Carbons than are in the molecule)
+            if (_formulaBox.NeutralFormula != null && _childAdducts != null)
+            {
+                foreach (var childAdduct in _childAdducts)
+                {
+                    try
+                    {
+                        childAdduct.ApplyToFormula(_formulaBox.NeutralFormula);
+                    }
+                    catch (InvalidOperationException e)

```

Comment:
"ApplyToFormula" can also throw an ArgumentExcepion if there was an error parsing "_formulaBox.NeutralFormula".
So, if I type in a formula which has more than one minus sign in it, then an unhandled ArgumentException might be thrown.

You might want to move this code to some point after you have already made sure that "_formulaBox.NeutralFormula" is a valid formula.
Or maybe change this to catch all exceptions.

Reply:
Excellent point. I always appreciate your reviews.